libyul/optimiser/StructuralSimplifier.cpp

```diff
@@ -202,10 +205,11 @@ void StructuralSimplifier::simplify(std::vector<yul::Statement>& _statements)
 		_statements,
 		[&](Statement& _stmt) -> OptionalStatements
 		{
-			visit(_stmt);
 			OptionalStatements result = boost::apply_visitor(visitor, _stmt);
 			if (result)
 				simplify(*result);
+			else
+				visit(_stmt);
```

Comment:
Should we put this in its own PR and give it a bugfix changelog entry? I mean: it could even incorrectly change the semantics of code before... Actually given #5959 I'd even consider it for the buglist...

Reply:
Ah nevermind, it has at least a Changelog entry here, I missed that.

Reply:
-> #6207

Reply:
Ah :-). I should read PR descriptions :-).